### PR TITLE
fix: Default to Daily if Viable

### PR DESCRIPTION
**Pull request by joeauyeung**
## What does this PR do?

This PR fixes a bug where an event type would default to Daily even if it would throw an error.

Fixes # (issue)

## Requirement/Documentation

- If there is a requirement document, please, share it here.
- If there is ab UI/UX design document, please, share it here.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- Create a new event type, do not set the location
- Create a booking under that new event type
    - The `EventManager` should not throw an error

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by emrysal — CHANGES_REQUESTED** [review]
image

File: packages/core/EventManager.ts [hunk 1]
```diff
@@ -94,7 +95,22 @@ export default class EventManager {
...
+        },
+        select: {
+          keys: true,
+          enabled: true,
+        },
+      });
+
+      const calVideoKeys = calVideoKeysSchema.safeParse(calVideo?.keys);
+
+      if (calVideo?.enabled && calVideoKeys.success) evt["location"] = "integrations:daily";
```
Comment (joeauyeung) [inline packages/core/EventManager.ts:112]: Add Zod safeParse to ensure that the keys are correct, and that means we don't have to do weird type hacks for `Prisma.JsonValue`

@emrysal

**Review by emrysal — DISMISSED** [review]
Works!

**Comment by joeauyeung** [comment]
> @joeauyeung e2e tests are failing and it is related to your changes

Tests are failing because the result is expecting Daily as the video location. Should I change the tests to not expect it?

**Comment by CarinaWolli** [comment]
> Tests are failing because the result is expecting Daily as the video location. Should I change the tests to not expect it?

Yes, let's change that

**Review by CarinaWolli — APPROVED** [review]

**Pull request merged**